packages/engine/src/scene/functions/createMap.ts

```diff
@@ -19,4 +21,20 @@ export async function createMap(entity: Entity, args: MapProps): Promise<void> {
   if (args.enableDebug) {
     addComponent(entity, DebugNavMeshComponent, null)
   }
+  currentEnt = entity
+}
+
+export async function updateMap(args: MapProps, longtitude, latitude, position): Promise<void> {
+  const remobj = Engine.scene.getObjectByName('MapObject')
+  const { mapMesh, navMesh, groundMesh } = await update(Engine.renderer, args, longtitude, latitude, position)
+
+  remobj.removeFromParent()
+  Engine.scene.add(mapMesh)
+
+  // getComponent(currentEnt, Object3DComponent).value.clear()
+  // getComponent(currentEnt, Object3DComponent).value.add(mapMesh)
```

Comment:
Tried reassigning to `component.value` i.e. `getComponent(currentEnt, Object3DComponent).value = mapMesh`?

Reply:
tried it. not works